src/types/addresses.ts

```diff
@@ -0,0 +1,15 @@
+export type ContractAddress<chainId = number> = { address: `0x${string}`; chainId: chainId; blockCreated?: number }
+export type Addresses<chainId = number> = {
+  portal: ContractAddress<chainId>
+  l2OutputOracle: ContractAddress<chainId>
+  l1StandardBridge: ContractAddress<chainId>
+  l1CrossDomainMessenger: ContractAddress<chainId>
+  l1Erc721Bridge: ContractAddress<chainId>
+}
+
+export type RawOrContractAddress<chainId> = `0x${string}` | ContractAddress<chainId>
```

Comment:
```suggestion
import {Address} from 'viem'
export type ContractAddress<chainId = number> = { address: `0x${string}`; chainId: chainId; blockCreated?: number }
export type Addresses<chainId = number> = {
  portal: ContractAddress<chainId>
  l2OutputOracle: ContractAddress<chainId>
  l1StandardBridge: ContractAddress<chainId>
  l1CrossDomainMessenger: ContractAddress<chainId>
  l1Erc721Bridge: ContractAddress<chainId>
}

export type RawOrContractAddress<chainId> = Address | ContractAddress<chainId>
```
We want to consistently use the type from viem or abitype. The reason is because the user can actually configure the address type to be string or something different if they choose. https://abitype.dev/config#addresstype

Reply:
Thank U! Changed this